### ISSUE TITLE
History navigation: calendar, or enter date

### ISSUE THREAD

**Issue by nadjet09** [nextcloud/maps#1222]
The slide bar for history navigation is useful if user is navigation same day, but when dealing with large tracking history, it's become a pain and time consuming to go to another date. You'll have to keep sliding and sliding may 20 ~ 30 time in order to get to a diffrent date that is not 2 days far from your selection.

I suggest to mainly use calendar to select the date range and inside that date range you can use the slide bar for a fine filtering and navigation. 

Please and thank you.

**Comment by nadjet09** [comment]
Just wanted to add more context about the navigation bar, hope this image makes it more clear.

![Maps - Nextcloud](https://github.com/nextcloud/maps/assets/48478986/8bec5f01-ee4f-48d3-a0ec-3b1aafa2201b)

**Comment by tacruc** [comment]
I agree, a rework of the filter would be nice. Unfortunately, my PhD thesis eats my time. 
If anyone wants to give it a try I can help.

**Comment by nadjet09** [comment]
thank you so much for considering it, and good luck with your PhD

**Comment by nadjet09** [comment]
Just wanted to upvote this, to see if it's still under consideration, it's useful to add this feature to history navigation of the location trackings

**Comment by tacruc** [comment]
Sorry unfortunately still out of time.

**Comment by nadjet09** [comment]
No worries, I've cloned  tried to paly abit with the source code, I have some c/cpp embedded system background, but never did any Javascript programming, and felt a bit lost to reverse engineering your code, do you have any recommendations for me where should I start? I can continue with the help of claude ai or chatgpt to make it work at the end.

**Comment by tacruc** [comment]
I'm currently on my phone. So exuseexcuse Me for the bad formatted message. Here are some quick helpful links here.
You probably want to get some basic knowledge about vue. 
https://vuejs.org/guide/quick-start

The frontend is written in vue2 (which is outdated). 

The slider component is implemented in:
https://github.com/nextcloud/maps/blob/master/src/components/map/Slider.vue

The https://github.com/nextcloud/maps/blob/de794addfc8f6b53ae45aa937887bd95030dfe10/src/views/App.vue#L267 binds all components together. Here the sliderStart and sliderEnd stores the selected range.

For building the Vue App you need the two commands.
`npm build`
And
`npm run watch`

**Comment by nadjet09** [comment]
Thank you so much, rhe instructions are clear, I'll take a look on vuejs

**Comment by tacruc** [comment]
By the way: https://github.com/juliusknorr/nextcloud-docker-dev 
is super helpful for a development setup.
